### PR TITLE
add ALPM source

### PR DESCRIPTION
**Pull request by DDoSolitary**

### CODE REVIEW

**Comment by DDoSolitary** [comment]
TODO: Tests and documentation

**Comment by DDoSolitary** [comment]
Should I add `pyalpm` to `install_requires` in `setup.py`?

**Comment by lilydjwg** [comment]
> Should I add pyalpm to install_requires in setup.py?

No. It's optional; who uses this source will need to install it.

File: tests/test_alpm.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  pkg_file = subprocess.check_output(['makepkg', '--packagelist'], cwd=pkg_dir, text=True).strip()
+  db_dir = os.path.join(temp_dir.name, 'test-db')
+  os.mkdir(db_dir)
+  repo_dir = os.path.join(db_dir, 'sync')
+  os.mkdir(repo_dir)
+  subprocess.check_call([
+    'repo-add',
+    os.path.join(repo_dir, 'test-repo.db.tar.gz'),
+    os.path.join(pkg_dir, pkg_file)
+  ])
```
Comment (lilydjwg) [inline tests/test_alpm.py:42]: Too many `os.path`s. Try to use `pathlib`?

File: nvchecker_source/alpm.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from pyalpm import Handle
+
+async def get_version(name, conf, **kwargs):
+  pkgname = conf.get('alpm', name)
+  dbpath = conf.get('dbpath', '/var/lib/pacman')
+  repo = conf.get('repo')
+  strip_release = conf.get('strip_release', False)
+  provided = conf.get('provided')
+  handle = Handle('/', dbpath)
+  db = handle.register_syncdb(repo, 0)
```
Comment (lilydjwg) [inline nvchecker_source/alpm.py:14]: Please cache the `db` or similar using the `cache` argument passed in.

**Review by lilydjwg — CHANGES_REQUESTED** [review]

File: nvchecker_source/alpm.py (hunk 2 above)
Comment (DDoSolitary) [reply nvchecker_source/alpm.py:14]: Done.

File: tests/test_alpm.py (hunk 1 above)
Comment (DDoSolitary) [reply tests/test_alpm.py:44]: Done.

File: nvchecker_source/alpm.py (hunk 2 above)
Comment (DDoSolitary) [reply nvchecker_source/alpm.py:14]: Should I cache the result of `get_pkg`? It seems that it loops through the package cache to find the specified name. I'm not sure if it should be considered an expensive operation.

Comment (DDoSolitary) [reply nvchecker_source/alpm.py]: Done.

**Pull request merged**